crates/apub/src/http/mod.rs

```diff
@@ -136,38 +136,35 @@ async fn check_community_content_fetchable(
 ) -> LemmyResult<()> {
   use CommunityVisibility::*;
   check_community_removed_or_deleted(community)?;
-  match community.visibility {
-    // content in public community can always be fetched
-    Public => Ok(()),
-    // no federation for local only community
-    LocalOnly => Err(LemmyErrorType::NotFound.into()),
-    // for private community check http signature of request, if there is any approved follower
-    // from the fetching instance then fetching is allowed
-    Private => {
-      let signing_actor = signing_actor::<SiteOrCommunityOrUser>(request, None, context).await?;
-      if community.local {
-        Ok(
-          CommunityFollowerView::check_has_followers_from_instance(
-            community.id,
-            signing_actor.instance_id(),
-            &mut context.pool(),
-          )
-          .await?,
+  if !community.visibility.can_federate() {
+    return Err(LemmyErrorType::NotFound.into());
+  }
+  if community.visibility == Private {
+    let signing_actor = signing_actor::<SiteOrCommunityOrUser>(request, None, context).await?;
```

Comment:
I think this whole block would be a lot clearer to just use `match community.visibility`, and handle all the cases.

Reply:
Changed